modules/component-store/src/component-store.ts

```diff
@@ -322,7 +322,7 @@ export class ComponentStore<T extends object> implements OnDestroy {
    * but not used with provideComponentStore()
    */
   private checkProviderForHooks() {
-    asyncScheduler.schedule(() => {
+    asapScheduler.schedule(() => {
```

Comment:
Thanks for the PR, @amakhrov! Can you add a unit test that verifies that there is no error?

Reply:
what if I just change `tick(0)` to `flushMicrotasks()` in the existin tests (there are 2 of them for the hooks feature).

Reply:
sounds good 👍

Reply:
@timdeschryver It's hard to test for that. I'd say the `flushMicrotasks();` change in unit tests is the one that "reproduces" the issue.